src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) throws RuntimeException {
+            String fruitName = transaction.getFruit();
+            int oldQuantity = storage.get(fruitName);
+            int newQuantity = oldQuantity - transaction.getQuantity();
+            if (newQuantity < 0) {
```

Comment:
Handling the Purchase operation correctly according to the checklist: the check for a negative balance is implemented properly.